### ISSUE TITLE
Errors in data augmentation

### ISSUE THREAD

**Issue by muyuan-ma** [ultralytics/ultralytics#18288]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

When I was inspecting the train_batch images, I noticed that some images were missing their labels. I’m not yet sure what caused this issue, but it seems that labels are lost specifically for images that have undergone flip data augmentation. As shown in the figure, one image includes my original image and its labels, while in the train_batch images, I captured a section where the labels appear to be missing. Here is my training code, which I ran directly.
<img width="573" alt="Snipaste_2024-12-18_12-09-21" src="https://github.com/user-attachments/assets/20bea981-d2da-447d-b976-6eb7f2091167" />
<img width="334" alt="Snipaste_2024-12-18_12-10-58" src="https://github.com/user-attachments/assets/d18ac06f-de83-4970-af90-37584f0a1c1d" />

### Environment

Ultralytics 8.3.32 🚀 Python-3.9.20 torch-2.5.1+cu118 CUDA:0 (NVIDIA A100-PCIE-40GB, 40514MiB)
Setup complete ✅ (40 CPUs, 240.0 GB RAM, 2.9/48.9 GB disk)

OS                  Linux-5.4.0-100-generic-x86_64-with-glibc2.27
Environment         Linux
Python              3.9.20
Install             git
RAM                 240.00 GB
Disk                2.9/48.9 GB
CPU                 Intel Xeon Silver 4316 2.30GHz
CPU count           40
GPU                 NVIDIA A100-PCIE-40GB, 40514MiB
GPU count           4
CUDA                11.8

numpy               ✅ 1.26.3>=1.23.0
numpy               ✅ 1.26.3<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.9.2>=3.3.0
opencv-python       ✅ 4.9.0.80>=4.6.0
pillow              ✅ 10.2.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.13.1>=1.4.1
torch               ✅ 2.5.1+cu118>=1.8.0
torch               ✅ 2.5.1+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.20.1+cu118>=0.9.0
tqdm                ✅ 4.67.0>=4.64.0
psutil              ✅ 5.9.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.11>=2.0.0
{'OS': 'Linux-5.4.0-100-generic-x86_64-with-glibc2.27', 'Environment': 'Linux', 'Python': '3.9.20', 'Install': 'git', 'RAM': '240.00 GB', 'Disk': '2.9/48.9 GB', 'CPU': 'Intel Xeon Silver 4316 2.30GHz', 'CPU count': 40, 'GPU': 'NVIDIA A100-PCIE-40GB, 40514MiB', 'GPU count': 4, 'CUDA': '11.8', 'Package Info': {'numpy': '✅ 1.26.3<2.0.0; sys_platform == "darwin"', 'matplotlib': '✅ 3.9.2>=3.3.0', 'opencv-python': '✅ 4.9.0.80>=4.6.0', 'pillow': '✅ 10.2.0>=7.1.2', 'pyyaml': '✅ 6.0.2>=5.3.1', 'requests': '✅ 2.32.3>=2.23.0', 'scipy': '✅ 1.13.1>=1.4.1', 'torch': '✅ 2.5.1+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"', 'torchvision': '✅ 0.20.1+cu118>=0.9.0', 'tqdm': '✅ 4.67.0>=4.64.0', 'psutil': '✅ 5.9.0', 'py-cpuinfo': '✅ 9.0.0', 'pandas': '✅ 2.2.3>=1.1.4', 'seaborn': '✅ 0.13.2>=0.11.0', 'ultralytics-thop': '✅ 2.0.11>=2.0.0'}}

### Minimal Reproducible Example

```
from ultralytics import YOLO
import wandb

wandb.login(key="keys")


model = YOLO("/ultralytics/cfg/models/11/yolo11-obb.yaml")
model = YOLO("/weights/yolo11l-obb.pt")  # load a pretrained model (recommended for training)

# 定义训练参数，添加默认值、范围和中文注释
train_params = {
    'project': None,            # 项目名称，保存训练结果的目录，默认值 None
    'name': '',            # 训练运行的名称，用于创建子目录保存结果，默认值 None
    'data': "",   # 数据集配置文件路径，需要自定义修改 coco128.yaml, coco8.yaml, dota8.yaml
    'imgsz': 1024,               # 输入图像大小，默认值 1024，范围 >= 32
    'epochs': 1000,               # 总训练轮次，默认值 100，范围 >= 1
    'device': 'cuda:2,3',             # 训练设备，默认值 None，支持 "cpu", "gpu"(device=0,1), "mps"
    'patience':500,
    'batch': 32,                # 批次大小，默认值 16，范围 >= 1
    'save': True,               # 是否保存训练结果和模型，默认值 True
    'save_period': -1,          # 模型保存频率，默认值 -1，表示只保存最终结果
    'cache': True,             # 是否缓存数据集，默认值 False
    'workers': 8,               # 数据加载线程数，默认值 8，影响数据预处理速度
    'exist_ok': False,          # 是否覆盖已有项目/名称目录，默认值 False
    'pretrained': True,
    'optimizer': 'auto',        # 优化器，默认值 'auto'，支持 'SGD', 'Adam', 'AdamW'
    'verbose': True,            # 是否启用详细日志输出，默认值 False
    'seed': 0,                  # 随机种子，确保结果的可重复性，默认值 0
    'deterministic': True,      # 是否强制使用确定性算法，默认值 True
    'single_cls': False,        # 是否将多类别数据集视为单一类别，默认值 False
    'rect': False,              # 是否启用矩形训练（优化批次图像大小），默认值 False
    'cos_lr': False,            # 是否使用余弦学习率调度器，默认值 False
    'close_mosaic': 100,         # 在最后 N 轮次中禁用 Mosaic 数据增强，默认值 10
    'resume': False,            # 是否从上次保存的检查点继续训练，默认值 False
    'amp': True,                # 是否启用自动混合精度（AMP）训练，默认值 True
    'fraction': 1.0,            # 使用数据集的比例，默认值 1.0
    'profile': False,           # 是否启用 ONNX 或 TensorRT 模型优化分析，默认值 False
    'freeze': None,             # 冻结模型的前 N 层，默认值 None
    'lr0': 0.01,                # 初始学习率，默认值 0.01，范围 >= 0
    'lrf': 0.01,                # 最终学习率与初始学习率的比值，默认值 0.01
    'momentum': 0.937,          # SGD 或 Adam 的动量因子，默认值 0.937，范围 [0, 1]
    'weight_decay': 0.0005,     # 权重衰减，防止过拟合，默认值 0.0005
    'warmup_epochs': 3.0,       # 预热学习率的轮次，默认值 3.0
    'warmup_momentum': 0.8,     # 预热阶段的初始动量，默认值 0.8
    'warmup_bias_lr': 0.1,      # 预热阶段的偏置学习率，默认值 0.1
    'box': 7.5,                 # 边框损失的权重，默认值 7.5
    'cls': 0.5,                 # 分类损失的权重，默认值 0.5
    'dfl': 0.75,                 # 分布焦点损失的权重，默认值 1.5
    'pose': 12.0,               # 姿态损失的权重，默认值 12.0
    'kobj': 1.0,                # 关键点目标损失的权重，默认值 1.0
    'label_smoothing': 0.0,     # 标签平滑处理，默认值 0.0
    'nbs': 64,                  # 归一化批次大小，默认值 64
    'overlap_mask': True,       # 是否在训练期间启用掩码重叠，默认值 True
    'mask_ratio': 4,            # 掩码下采样比例，默认值 4
    'dropout': 0.0,             # 随机失活率，用于防止过拟合，默认值 0.0
    'val': True,                # 是否在训练期间启用验证，默认值 True
    'plots': True,             # 是否生成训练曲线和验证指标图，默认值 True
 
    # 数据增强相关参数
    'hsv_h': 0.015,             # 色相变化范围 (0.0 - 1.0)，默认值 0.015
    'hsv_s': 0.7,               # 饱和度变化范围 (0.0 - 1.0)，默认值 0.7
    'hsv_v': 0.4,               # 亮度变化范围 (0.0 - 1.0)，默认值 0.4
    'degrees': 180.0,             # 旋转角度范围 (-180 - 180)，默认值 0.0
    'translate': 0.1,           # 平移范围 (0.0 - 1.0)，默认值 0.1
    'scale': 0.5,               # 缩放比例范围 (>= 0.0)，默认值 0.5
    'shear': 5,               # 剪切角度范围 (-180 - 180)，默认值 0.0
    'perspective': 0.0,         # 透视变化范围 (0.0 - 0.001)，默认值 0.0
    'flipud': 0.5,              # 上下翻转概率 (0.0 - 1.0)，默认值 0.0
    'fliplr': 0.5,              # 左右翻转概率 (0.0 - 1.0)，默认值 0.5
    'bgr': 0.0,                 # BGR 色彩顺序调整概率 (0.0 - 1.0)，默认值 0.0
    'mosaic': 0.5,              # Mosaic 数据增强 (0.0 - 1.0)，默认值 1.0
    'mixup': 0.05,               # Mixup 数据增强 (0.0 - 1.0)，默认值 0.0
    'copy_paste': 0.0,          # Copy-Paste 数据增强 (0.0 - 1.0)，默认值 0.0
    'copy_paste_mode': 'flip',  # Copy-Paste 增强模式 ('flip' 或 'mixup')，默认值 'flip'
    'auto_augment': 'randaugment',  # 自动增强策略 ('randaugment', 'autoaugment', 'augmix')，默认值 'randaugment'
    'erasing': 0.2,             # 随机擦除增强比例 (0.0 - 0.9)，默认值 0.4
    'crop_fraction': 1.0,       # 裁剪比例 (0.1 - 1.0)，默认值 1.0
}


# Train the model
results = model.train(**train_params)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @muyuan-ma, thank you for bringing this to our attention and for sharing such detailed information and diagrams! 🚀 We appreciate your interest in Ultralytics.

To help us address this issue, could you provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum_reproducible_example/) that demonstrates the behavior you're reporting? This will allow us to better debug and identify the root cause. If you're observing this during training, it may also be helpful to confirm whether the issue persists when using alternative datasets or configurations.

In the meantime, here are a few steps you can follow to ensure your environment and setup are up-to-date:
  
## Upgrade

Make sure you're running the latest version of the `ultralytics` package in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Verified Environments

YOLO runs seamlessly in the following environments. If needed, you can explore these to reproduce your issue with everything pre-configured:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Documentation and Resources

While waiting, we also recommend exploring these guides for potential solutions or adjustments to your training settings:
- [Docs](https://docs.ultralytics.com) for full library usage guides
- [Model Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) to improve training performance
- [CLI Usage](https://docs.ultralytics.com/usage/cli/) and [Python Usage](https://docs.ultralytics.com/usage/python/) examples

## Need Support?

Engage with the community for faster assistance and shared experiences:
- Join our [Discord](https://discord.com/invite/ultralytics) for direct conversations 🎧  
- Post on [Discourse](https://community.ultralytics.com) for detailed discussions  
- Check out our [Subreddit](https://reddit.com/r/Ultralytics) for community-driven advice  

⚠️ This is an automated response to streamline support, but rest assured that an Ultralytics engineer will review your issue and provide direct assistance soon. Thank you for your patience! 😊

**Comment by Y-T-G** [comment]
Can you post the txt labels for the image?

Does the logs say there are `backgrounds` in the dataset?

**Comment by muyuan-ma** [comment]
@Y-T-G 
1. I reran a new experiment with this dataset and encountered the same issue. Here are the txt labels for the image: 
```
4 0.9585290689528196 0.46078064440378286 1.0108527417452828 0.4247125589622641 0.9927303166287158 0.3984225069565191 0.9404066438362524 0.4344905923980377
1 0.6012843459752322 0.40839057662538697 0.607868477261502 0.3886663455115107 0.5003023413312694 0.3527597716718266 0.49371821004499955 0.3724840027857028
1 0.5620964593548715 0.48837272126334397 0.5684799522178825 0.4684132892466676 0.4623138632550336 0.4344588926174497 0.4559303703920226 0.454418324634126
4 0.9740192439344614 0.7234838377799143 1.0083976070487466 0.7044355263106981 0.9973655523255813 0.6845248909883721 0.9629871892112961 0.7035732024575883
1 0.5124058044606694 0.5541593323294527 0.5189552682310814 0.5331664689552169 0.3855740510767452 0.49155349630671674 0.3790245873063332 0.5125463596809525
1 0.3251174238914365 0.5871229306396809 0.4287003610108303 0.6226026624548736 0.42331862606155485 0.6383146164162994 0.31973568894216103 0.6028348846011066
1 0.37983740614199735 0.5532008965810868 0.4833455550541516 0.5898155460288809 0.4778088214938378 0.6054676702603317 0.3743006725816834 0.5688530208125375
1 0.4999595339637773 0.46123889886841474 0.6261524180178326 0.5087382602832928 0.6187246164259927 0.5284719088447654 0.4925317323719374 0.4809725474298874
0 0.3742613596386724 0.04501981200648456 0.8677610276826823 0.7969537779051976 0.5693873355263158 0.9927785773026316 0.07588766748230598 0.2408446114039185
```
2. The log shows: train: Scanning xxx/train.cache... 802 images, 0 backgrounds,

**Comment by Y-T-G** [comment]
Can you try installing the latest version using:

`pip install git+https://github.com/ultralytics/ultralytics`

**Comment by Laughing-q** [comment]
> @Y-T-G
> 
> 1. I reran a new experiment with this dataset and encountered the same issue. Here are the txt labels for the image:
> 
> ```
> 4 0.9585290689528196 0.46078064440378286 1.0108527417452828 0.4247125589622641 0.9927303166287158 0.3984225069565191 0.9404066438362524 0.4344905923980377
> 1 0.6012843459752322 0.40839057662538697 0.607868477261502 0.3886663455115107 0.5003023413312694 0.3527597716718266 0.49371821004499955 0.3724840027857028
> 1 0.5620964593548715 0.48837272126334397 0.5684799522178825 0.4684132892466676 0.4623138632550336 0.4344588926174497 0.4559303703920226 0.454418324634126
> 4 0.9740192439344614 0.7234838377799143 1.0083976070487466 0.7044355263106981 0.9973655523255813 0.6845248909883721 0.9629871892112961 0.7035732024575883
> 1 0.5124058044606694 0.5541593323294527 0.5189552682310814 0.5331664689552169 0.3855740510767452 0.49155349630671674 0.3790245873063332 0.5125463596809525
> 1 0.3251174238914365 0.5871229306396809 0.4287003610108303 0.6226026624548736 0.42331862606155485 0.6383146164162994 0.31973568894216103 0.6028348846011066
> 1 0.37983740614199735 0.5532008965810868 0.4833455550541516 0.5898155460288809 0.4778088214938378 0.6054676702603317 0.3743006725816834 0.5688530208125375
> 1 0.4999595339637773 0.46123889886841474 0.6261524180178326 0.5087382602832928 0.6187246164259927 0.5284719088447654 0.4925317323719374 0.4809725474298874
> 0 0.3742613596386724 0.04501981200648456 0.8677610276826823 0.7969537779051976 0.5693873355263158 0.9927785773026316 0.07588766748230598 0.2408446114039185
> ```
> 
> 2. The log shows: train: Scanning xxx/train.cache... 802 images, 0 backgrounds,

@muyuan-ma Hi! can you provide us a image as well to reproduce the issue if possible? Thanks!
